R/checkers.R

```diff
@@ -186,58 +186,42 @@ throw_type_error <- function(object,
 #' the concentration measurements in the wastewater data
 #' @param lod_col_name string indicating the name of the column containing
 #' the concentration measurements in the wastewater data
+#' @param add_req_col_names vector of strings indicating the required wastewater
+#' column names, the defaults are `c("date", "site", "lab", "site_pop")`
 #' @param call Calling environment to be passed to [cli::cli_abort()] for
 #' traceback.
 #'
 #' @return NULL, invisibly
 check_required_ww_inputs <- function(ww_data,
                                      conc_col_name,
                                      lod_col_name,
+                                     add_req_col_names = c(
+                                       "date", "site",
+                                       "lab", "site_pop"
+                                     ),
                                      call = rlang::caller_env()) {
   column_names <- colnames(ww_data)
-  if (!"date" %in% column_names) {
+  expected_col_names <- c(
+    {
+      conc_col_name
+    },
+    {
+      lod_col_name
+    },
+    add_req_col_names
+  )
+  res <- try(checkmate::assert_names(column_names,
+    must.include = expected_col_names
+  ))
+  if (class(res) == "try-error") {
     cli::cli_abort(
-      c("`date` column missing from input wastewater data"),
+      c("Required columns are missing from the input wastewater data"),
       class = "wwinference_input_data_error",
       call = call
     )
   }
```

Comment:
If you want to set a custom error message, the simplest thing to do here is use [`checkmate::check_names()`](https://mllg.github.io/checkmate/reference/checkNames.html) (which returns `TRUE` on check success and `FALSE` on check failure) rather than using `checkmate::assert_names` and a try/catch.

```suggestion
  if(!checkmate::check_names(column_names, must.include = expected_col_names)) {
    cli::cli_abort(
      c("`date` column missing from input wastewater data"),
      c("Required columns are missing from the input wastewater data"),
      class = "wwinference_input_data_error",
      call = call)
  }